File: baekjoon/greedy/11497/11497.java
import java.util.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.BufferedWriter;
import java.io.OutputStreamWriter;

public class Main{

  public static void main(String []args) throws Exception{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    PriorityQueue<Long> card = new PriorityQueue<>();

    int T = Integer.parseInt(br.readLine());
    for(int t=0;t<T;t++){
      int N = Integer.parseInt(br.readLine());
      PriorityQueue<Integer> arr = new PriorityQueue<>(Collections.reverseOrder());

      String[] tmp = br.readLine().split(" ");
      for(int i=0;i<N;i++){
        arr.offer(Integer.parseInt(tmp[i]));
      }
      
      int[] last = {arr.peek(), arr.poll()};
      int diff = 0;
      int for_len = 0;
      if(N%2==0) for_len = N/2 -1;
      else for_len = N/2;

      for(int i=0;i<for_len;i++){
        int[] pop = {arr.poll(), arr.poll()};
        diff = Collections.max(Arrays.asList(new Integer[]{diff, last[0]-pop[0], last[1]-pop[1]}));
        last[0] = pop[0]; last[1] = pop[1];
      }

      if(N%2==0){
        int pop = arr.poll();
        diff = Collections.max(Arrays.asList(new Integer[]{diff, last[0]-pop, last[1]-pop}));
      }

      bw.write(diff+"\n");
    }
    bw.flush();
    bw.close();
    br.close();
  } 
}

File: baekjoon/greedy/11497/11497.py
import sys, heapq

T = int(sys.stdin.readline())

for _ in range(T):
  N = int(sys.stdin.readline())
  arr = list(map(lambda x: -int(x), sys.stdin.readline().split()))
  heapq.heapify(arr)

  last = [heapq.heappop(arr)]*2
  diff = 0
  for_len = N//2 - 1 if N%2==0 else N//2

  for _ in range(for_len):
    pop = [heapq.heappop(arr), heapq.heappop(arr)]
    diff = max(diff, -last[0] + pop[0], -last[1] + pop[1])
    last = pop
  
  if N%2 == 0: 
    pop = heapq.heappop(arr)
    diff = max(diff, -last[0] + pop, -last[1] + pop)

  print(diff)


